api/src/services/Solr.ts

```diff
@@ -32,6 +32,18 @@ class Solr {
         return http(method, url, data, options);
     }
 
+    public async query(
+        core: string,
+        solrQuery: string,
+        queryParams: Record<string, string> = {}
+    ): Promise<NeedleResponse> {
+        let extras = "";
+        for (const i in queryParams) {
+            extras += "&" + encodeURIComponent(i) + "=" + encodeURIComponent(queryParams[i]);
```

Comment:
`Object.values(queryParams).reduce((extras, value) => { return extras + "";}),"");`

Reply:
I don't think a reduce operation can work here because we need to account for both keys and values... however, for a more functional twist on the logic, I tried this instead, which I think is an improvement: d1b5bd5